inference-engine/src/legacy_api/src/network_serializer.cpp

```diff
@@ -412,9 +415,128 @@ std::vector<CNNLayerPtr> TopologicalSort(const ICNNNetwork& network) {
     return ordered;
 }
 
+namespace {
+
+void FillXmlDocWithExecutionNGraph(const InferenceEngine::ICNNNetwork& network,
+                                   pugi::xml_document& doc) {
+    std::shared_ptr<const ngraph::Function> function = network.getFunction();
+    if (function == nullptr) {
+        THROW_IE_EXCEPTION << network.getName() << " does not represent ngraph::Function";
+    }
+
+    std::vector<std::shared_ptr<ngraph::Node>> ordered = function->get_ordered_ops();
+    pugi::xml_node netXml = doc.append_child("net");
+    netXml.append_attribute("name").set_value(network.getName().c_str());
+
+    pugi::xml_node layers = netXml.append_child("layers");
+
+    std::map<std::shared_ptr<ngraph::Node>, size_t> matching;
+    for (size_t i = 0; i < ordered.size(); i++) {
+        matching[ordered[i]] = i;
+    }
+
+    for (size_t i = 0; i < ordered.size(); ++i) {
```

Comment:
I think you can move `matching[ordered[i]] = i;` to this cycle and remove the previous one.